interfaces.go

```diff
@@ -0,0 +1,223 @@
+package failure_detector
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+)
+
+type KeyFunc func(obj interface{}) string
+
+// NewStoreFunc a func for creating WeightedEndpointStatus store per service
+type NewStoreFunc func(ttl time.Duration) WeightedEndpointStatusStore
+
+// EvaluateFunc a function to an external policy evaluator that sets the status and weight of the given endpoint based on the collected samples.
```

Comment:
what's the bool?

Reply:
"Evaluate" does not sound like it has side-effects.

Reply:
`bool` tells me if I should export status to the `readOnly` external store.

I was also thinking about detecting changes directly in `processBatch` func after calling `EvaluateFunc`